### ISSUE TITLE
[Tier 2] radixark/Megatron-LM@miles-main builds + MLP forward on gfx950

### ISSUE THREAD

**Issue by sunxxuns** [amdpilot-org/miles#9]
## Goal
Confirm the Megatron fork the AMD image depends on actually compiles its
fused kernels for gfx950 (not just gfx942) and a single MLP layer forwards.

## Steps
1. Inside the AMD image:
   ```bash
   git clone --branch miles-main https://github.com/radixark/Megatron-LM /tmp/megatron
   cd /tmp/megatron
   pip install -e .
   python -c "
   import torch
   from megatron.core.transformer.mlp import MLP
   from megatron.core.transformer.transformer_config import TransformerConfig
   cfg = TransformerConfig(
       num_layers=1, hidden_size=512, num_attention_heads=8,
       ffn_hidden_size=2048, use_cpu_initialization=True)
   m = MLP(cfg, submodules=None).cuda().bfloat16()
   x = torch.randn(2, 4, 512, device='cuda', dtype=torch.bfloat16)
   print('out', m(x)[0].shape)
   "
   ```

## Acceptance
- No "kernel not found for gfx950" or "unsupported arch" errors.
- Output shape is `(2, 4, 512)` and contains no NaN.
---
**Target arch**: gfx950 (MI355X)
**Base image**: `rlsys/miles:MI350-355-latest` (Docker Hub, pushed 2026-05-12, pulls cleanly)
**Test cluster**: n08-09 (MI355X), g46 (MI355X)
**Upstream parity**: this issue tracks behavior on amdpilot-org/miles, kept
in sync with `radixark/miles@main`. File a separate upstream issue only after
confirming repro on a clean checkout.

Parent: #2

**Comment by jhinpan** [comment]
**AMDPilot — accumulated progress on this issue**

_Auto-generated from successive trials. Latest update: 2026-05-15T13:50:06Z by `issue-miles-9`._

### Verified infra fixes (apply during Stage 0)

- Add liveness probe to DockerManager entrypoint to crash-restart if no stdout within 300s


### Working approaches and their best metric

_(none yet — when a trial measurably improves the metric, the change lands here.)_


### Approaches that did NOT help — don't retry these

- Waiting 2960s for global timeout before reaping silent trials


### Open ideas to try next

- Debug entrypoint script for deadlocks or infinite loops before first status update
- Verify tool call initialization does not hang on mia1-p02-g05


_amdpilot bot. Edit between the markers to add/correct fixes; the next update will preserve your changes. To disable updates on this issue, replace the body with the marker line only._

**Comment by sunxxuns** [comment]
Closing per xsun (cycle 12 audit): this is verification/measurement work that doesn't require source-code or kernel changes — not worth burning agent + GPU lease overhead. The orchestrator is now restricted to issues that demand actual code work (kernel ports, HIP/MFMA rewrites, source patches). The remaining open issues (#18, #19, #23) are the real porting work.
